testing/resource.go

```diff
@@ -65,11 +66,29 @@ func (c *Resource) SetDefaults(ctx context.Context) {
 	c.Spec.SetDefaults(ctx)
 }
 
+type onContextKey struct{}
+
+// WithValue returns a WithContext for attaching an OnContext with the given value.
+func WithValue(ctx context.Context, val string) context.Context {
+	return context.WithValue(ctx, onContextKey{}, &OnContext{Value: val})
```

Comment:
double check this, but if you look at the docs for context.WithValue they imply that you should use a global `var onKey = onContextKey{}` and then use it `return context.WithValue(ctx, onKey, &OnContext{Value: val})`

I don't know if their internal impl cares.

Reply:
it doesn't, I copied this from our config store logic

Reply:
it might not matter.